pandas/core/groupby.py

```diff
@@ -2883,7 +2883,12 @@ def aggregate(self, func_or_funcs, *args, **kwargs):
             except Exception:
                 result = self._aggregate_named(func_or_funcs, *args, **kwargs)
 
-            index = Index(sorted(result), name=self.grouper.names[0])
+            # mixed types fail to sort
+            try:
+                values = sorted(result)
+            except TypeError:
+                values = result
```

Comment:
I tried this and it resulted in the same tests failing, not sure what's going on.